### PR TITLE
Composable.compose gets source as kwarg, which is equal to self

### PR DESCRIPTION
**Pull request by AlirezaSohofi**
# Description

Not having access to `source` in the `__init__` method when composing iterables makes it difficult to track steps of the pipeline (see #64). 

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. 
List any dependencies that are required for this change.

Fixes # issue

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring including code style reformatting 
- [ ] Other (please describe):

# Checklist:

- [ ] I have read the [contributing guideline doc](https://squirrel-core.readthedocs.io/en/latest/usage/code_of_conduct.html) (external contributors only)
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have kept the PR small so that it can be easily reviewed  
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] All dependency changes have been reflected in the pip requirement files.

### CODE REVIEW

**Review by pzdkn — DISMISSED** [review]
thanks :) LGTM!

File: squirrel/iterstream/base.py [hunk 1]
```diff
@@ -140,106 +226,14 @@ def monitor(
...
 class _Iterable(Composable):
     """
     A class representing an iterable, which applies the callable `f` to the `source` items and returns the result in
     the :py:meth:`__iter__` method. This is used as the object being passed between steps of the stream.
     """
 
-    def __init__(
-        self, source: t.Iterable, f: t.Callable[..., t.Iterator], *args, _kwa: t.Optional[t.Dict] = None, **kw
-    ):
+    def __init__(self, source: t.Iterable, f: t.Callable[..., t.Iterator], *args, **kw):
```
Comment (AlpAribal) [inline squirrel/iterstream/base.py:236]: Could you update the docstring as well?

File: squirrel/iterstream/source.py [hunk 2]
```diff
@@ -40,7 +40,7 @@ def __iter__(self) -> t.Iterator:
             yield from self.source
 
 
-class FilePathGenerator(IterableSource):
+class FilePathGenerator(Composable):
     """
     A specialized version of IterableSource that accepts a url without instantiating it eagerly. It simply generates
```
Comment (AlpAribal) [inline squirrel/iterstream/source.py:45]:
```suggestion
    A specialized version of Composable that accepts a url without instantiating it eagerly. It simply generates
```

**Review by AlpAribal — COMMENTED** [review]
Neat, nice catch!

It was rather hard to understand the diff since methods were reordered (IMO updating the compose/source_ methods here and doing the ordering in a separate PR, which we could merge to this one, would be easier for review).

**Review by pzdkn — APPROVED** [review]
LG!

File: squirrel/__init__.py [hunk 3]
```diff
@@ -1 +1 @@
-__version__ = "0.14.3"
+__version__ = "0.14.4"
```
Comment (AlpAribal) [inline squirrel/__init__.py:1]: Shall we have a minor bump (since `.source_()` got removed)?

**Pull request merged**